Actions/AL-Go-Helper.ps1

```diff
@@ -666,6 +666,8 @@ function ReadSettings {
             "Account"                                   = ""
             "CertificateProfile"                        = ""
         }
+        "useSubModules"                                 = ""
```

Comment:
```suggestion
        "useSubModules"                                 = $false
```

Reply:
useSubModules can be `true` to checkout submodules or `recursive` to recursively checkout submodules - a boolean would not suffice

Reply:
Yeah, so the options should be:
* True
* False (Default)
* Recursive

Reply:
Why would "false" be better than "" as undefined?
"false" is an arbitrary value, which cannot be given to actions/checkout - why is it better than "" - it isn't a value people will ever see.

Reply:
Defaulting to "false" would mean the code would look like:

```
      - name: Checkout Submodules
        if: env.useSubModules != 'false' && env.useSubModules != ''
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
```

as most other string settings in AL-Go uses an empty string as the default (I think) - but if you feel strongly about this - then fine.

Reply:
> Why would "false" be better than "" as undefined? "false" is an arbitrary value, which cannot be given to actions/checkout - why is it better than "" - it isn't a value people will ever see.

I stand corrected - 'false' can be given to actions/checkout - but we would never do that. I can however see that if anybody reads the documentation for actions/checkout - then they might assume that 'false' is a good value.

Reply:
fixed